node/tests/reconstruct_test.go

```diff
@@ -66,9 +69,13 @@ func TestFullReconstructFromBridge(t *testing.T) {
 			return full.ShareServ.SharesAvailable(bctx, h.DAH)
 		})
 	}
-
-	err = errg.Wait()
-	require.NoError(t, err)
+	select {
+	case <-ctx.Done():
+		t.Fatal("timeout reached")
+	case err = <-errCh:
+		require.NoError(t, err)
+	}
+	require.NoError(t, errg.Wait())
```

Comment:
Can we find a way to hide this behind swamp so that each test does not need to repeat the same logic and worry about handling the error?

Reply:
I can simplify it as much as possible. But we still have to handle error in main goroutine. Passing `t` instance to a separate goroutine makes test flakiness